app/Repositories/BaseRepository.php

```diff
@@ -60,6 +66,76 @@ public function archive($entity)
+                    ->exists();
+
+            if ($subscriptions) {
+                WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_QUOTE, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+            }}
+            elseif ($entity instanceof Credit){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_CREDIT)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_CREDIT, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
+            }
+            elseif ($entity instanceof Client){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_CLIENT)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_CLIENT, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
+            }
+            elseif ($entity instanceof Expense){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_EXPENSE)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_EXPENSE, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
+            }
+            elseif ($entity instanceof Project){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_PROJECT)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_PROJECT, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
+            }
+            elseif ($entity instanceof Task){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_TASK)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_TASK, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
+            }
+            elseif ($entity instanceof Payment){
+                $subscriptions = Webhook::where('company_id', $entity->company_id)
+                    ->where('event_id', Webhook::EVENT_ARCHIVE_PAYMENT)
+                    ->exists();
+
+                if ($subscriptions) {
+                    WebhookHandler::dispatch(Webhook::EVENT_ARCHIVE_PAYMENT, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
+                }
```

Comment:
I think we should extract this block to a separate method.

To avoid the long list of if/elseif perhaps use a switch or match block ie

```php
case($entity)
($entity instanceOf Invoice)
$event = Webhook::EVENT_ARCHIVE_INVOICE
break;
```
and then have a single query

```php

 $subscriptions = Webhook::where('company_id', $entity->company_id)
                    ->where('event_id', $event)
                    ->exists();

                if ($subscriptions) {
                    WebhookHandler::dispatch($event, $entity, $entity->company, 'client')->delay(now()->addSeconds(2));
                }
```

 Also you'll want to consider the includes are correct for each entity, you've included 'client' for all of them, but 'client' may not be appropriate for all of them.